js/web/lib/wasm/jsep/backend-webgpu.ts

```diff
@@ -378,14 +472,22 @@ export class WebGpuBackend {
       this.programManager.setArtifact(key, artifact);
       LOG_DEBUG('info', () => `[artifact] key: ${key}, programName: ${program.name}`);
     }
+    // update kernels
+    const kernelInfo = this.kernels.get(this.currentKernelId!)!;
```

Comment:
There is no guarentee that each kernel will run no more than once between two `flush()` calls.

This also makes `kernelInfo` data structure combined with const and mutable data.

I would suggest to simply use an array of { programName, inputs, outputs } to track those data.

Reply:
I think the kernelId are different if they're different nodes in the graph, even if they're same opType or programName. Do you mean the graph that has loop inside?
Anyway, I split the info into 2 parts, following your comments here. PTAL.